### PR TITLE
feat: smart router

### PR DESCRIPTION
**Pull request by chefjackson**

### CODE REVIEW

File: packages/smart-router/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "prepublishOnly": "yarn run build",
+    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
+  },
+  "dependencies": {
+    "@pancakeswap/farms": "*",
+    "@pancakeswap/multicall": "*",
+    "@pancakeswap/sdk": "*",
+    "@pancakeswap/tokens": "*",
+    "@pancakeswap/token-lists": "*",
+    "ethers": "^5.7.2",
```
Comment (chef-jojo) [inline packages/smart-router/package.json:22]: might need to fix the version, better split `ethers`

File: packages/smart-router/evm/getStableSwapPairs.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { ChainId, CurrencyAmount, Pair } from '@pancakeswap/sdk'
+import { getStableConfig } from '@pancakeswap/farms/constants'
```
Comment (chef-jojo) [inline packages/smart-router/evm/getStableSwapPairs.ts:2]: shouldn't rely on farms config 🤔 
that might be a case that stableswap pair is not in farms
let's add a constants for stableswap

File: packages/smart-router/evm/getStableSwapPairs.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { deserializeToken } from '@pancakeswap/token-lists'
+
+import { StableSwapPair } from './types'
+import { createStableSwapPair } from './stableSwap'
+
+export async function getStableSwapPairs(chainId: ChainId): Promise<StableSwapPair[]> {
+  // Stable swap is only supported on BSC chain
+  if (chainId !== ChainId.BSC) {
```
Comment (chef-jojo) [inline packages/smart-router/evm/getStableSwapPairs.ts:10]: can add testnet constants

File: packages/smart-router/package.json (hunk 1 above)
Comment (chefjackson) [reply packages/smart-router/package.json:22]: OK.

File: packages/smart-router/evm/getStableSwapPairs.ts (hunk 2 above)
Comment (chefjackson) [reply packages/smart-router/evm/getStableSwapPairs.ts:2]: 🤔 I thought we have farms for all pairs we supported. Let me add it's own config.

Comment (chefjackson) [reply packages/smart-router/evm/getStableSwapPairs.ts:2]: Do we have a lp config that maintains all token pairs that we support?

File: packages/smart-router/evm/getStableSwapPairs.ts (hunk 3 above)
Comment (chefjackson) [reply packages/smart-router/evm/getStableSwapPairs.ts]: done

File: packages/smart-router/package.json (hunk 1 above)
Comment (chefjackson) [reply packages/smart-router/package.json]: Done

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**